src/test/ui/parser/macro/issue-37113.rs

```diff
@@ -1,11 +0,0 @@
-macro_rules! test_macro {
-    ( $( $t:ty ),* $(),*) => {
-        enum SomeEnum {
-            $( $t, )* //~ ERROR expected identifier, found `String`
```

Comment:
This *should* be an error (at least with the current enum grammar).
It doesn't happen because the parser drops `Invisible` delimiters and turns this into `String` instead of `«String»`.

But in general some previously erroring code becoming legal is expected, because `$expr` can for example now be used in a type position if the underlying tokens match *both* expression and type grammars.